components/bellsound/BellSoundWalkthrough.tsx

```diff
@@ -0,0 +1,119 @@
+import { useState } from "react"
+import { Bike } from "../../lib/bike"
+import SelectFileStep from "./SelectFileStep"
+import UploadStep from "./UploadStep"
+import { Callout, CalloutKind } from "../Callouts"
+import WalkthroughButton from "./WalkthroughButton"
+import dynamic from "next/dynamic"
+
+const ConvertStep = dynamic(() => import("./ConvertStep"), { ssr: false })
+
+export type CommonProps = {
+    onDismiss: () => void
+}
+
+enum WalkthroughStep {
+    SelectFile,
+    Convert,
+    Upload,
+    Done
+}
+
+export default function BellSoundWalkthrough({ bike, onDismiss }: CommonProps & { bike: Bike }) {
+    const [currentStep, setCurrentStep] = useState<WalkthroughStep>(WalkthroughStep.SelectFile)
+    const [selectedFile, setSelectedFile] = useState<File | null>(null)
+    const [convertedFile, setConvertedFile] = useState<Uint8Array | null>(null)
+    const [error, setError] = useState<string | null>(null)
+
+    const onFileSelected = (file: File) => {
+        setError("")
+        setSelectedFile(file)
+        setCurrentStep(WalkthroughStep.Convert)
+    }
+
+    const onConversionCompleted = (convertedFile: Uint8Array) => {
+        setError("")
+
+        if (convertedFile.byteLength > 500_000) {
```

Comment:
```suggestion
        if (convertedFile.byteLength > 400_000) {
```

I think we should play it on the safe side and make it 400KB

Reply:
I don't want people get their bikes bricked and complain to us. (nor do I want my own bike to brick)